canvas_manage_course/views.py

```diff
@@ -21,8 +22,14 @@
 
 @require_http_methods(['GET'])
 def tool_config(request):
-    url = "%s://%s%s" % (request.scheme, request.get_host(),
-                         reverse('lti_launch', exclude_resource_link_id=True))
+
+    if request.is_secure():
+        host = 'https://' + request.get_host()
+    else:
+        host = 'http://' + request.get_host()
+
+    url = host + reverse('lti_launch')
```

Comment:
I would think that we'd always want our tool launch urls (this one, at least) to be over ssl.   IIRC, Canvas prefers/demands that LTI tools be running over secure connections - maybe it's actually browsers that enforce that due to mixed content warnings/errors.   In any case, I'd be tempted to simply make the url `https` here.